src/stories/form/StyledDropdown.tsx

```diff
@@ -0,0 +1,97 @@
+import { ControlProps } from './Form';
+import { useState } from 'react';
+import DfxIcon, { IconColors, IconSizes, IconVariant } from '../DfxIcon';
+import { Controller } from 'react-hook-form';
+
+export interface StyledDropdownProps extends ControlProps {
+  labelIcon?: IconVariant;
+  placeholder?: string;
+  items: DropdownItem[];
+  onSelect: (item: DropdownItem) => void;
+}
+
+export interface DropdownItem {
+  title: string;
+  description: string;
+}
+
+export default function StyledDropdown({
+  label,
+  labelIcon,
+  control,
+  name,
+  rules,
+  disabled,
+  items,
+  placeholder,
+  onSelect,
+  ...props
+}: StyledDropdownProps) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const select = (item: DropdownItem): void => {
+    setIsOpen(false);
+    onSelect(item);
+  };
+
+  let buttonClasses = 'flex justify-between border border-dfxGray-400 px-4 py-3 shadow-sm w-full';
+
+  isOpen ? (buttonClasses += ' rounded-x rounded-t bg-dfxGray-400/50') : (buttonClasses += ' rounded');
+
+  return (
+    <Controller
+      control={control}
+      render={({ field: { onChange, onBlur, value } }) => (
+        <div className="relative">
+          <div className="flex ml-3.5 mb-2.5">
+            {labelIcon !== undefined && <DfxIcon icon={labelIcon} size={IconSizes.SM} color={IconColors.BLUE} />}
+
+            <label className="text-dfxBlue-800 text-base font-semibold pl-3.5">{label}</label>
+          </div>
+          <button
+            id="dropDownButton"
+            type="button"
+            onClick={() => setIsOpen(!isOpen)}
+            className={buttonClasses}
+            onBlur={onBlur}
+            disabled={disabled}
+            {...props}
+          >
+            <div className="flex flex-col gap-2 justify-between text-left">
+              {value === undefined ? (
```

Comment:
Use only `==` for comparison to `undefined`, except you explicitly want `null` values.